react-native/react/menubar/index.js

```diff
@@ -40,12 +41,20 @@ class Menubar extends Component {
     const onMenubarShow = () => {
       setImmediate(() => {
         this.checkForFolders()
+
+        engine.listenOnConnect('menubar', () => {
```

Comment:
Unclear, this is all tricky as react is a little sensitive about when these types of functions are called, so if this is called as a side effect of something else you need to reason a lot about if its safe or not. it might be better to have a wrapper around ipcRenderer callbacks to always wrap it just in case.